src/v/storage/log_housekeeping_meta.h

```diff
@@ -13,8 +13,14 @@
 
 #include "storage/log.h"
 
+#include <boost/intrusive/list_hook.hpp>
+#include <boost/intrusive/options.hpp>
+
 namespace storage {
 struct log_housekeeping_meta {
+    using intrusive_link_type = boost::intrusive::list_member_hook<
+      boost::intrusive::link_mode<boost::intrusive::auto_unlink>>;
```

Comment:
we have a header `utils/intrusive_list_helpers.h` that provide typedefs for common cases that we use in the tree.